src/DataSyncInfrastructure/Doctrine/ORM/CustomersQueryBuilder.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Webgriffe\SyliusClerkPlugin\DataSyncInfrastructure\Doctrine\ORM;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Sylius\Bundle\CoreBundle\Doctrine\ORM\CustomerRepository;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Webgriffe\SyliusClerkPlugin\DataSyncInfrastructure\Doctrine\ORM\Event\QueryBuilderEvent;
+use Webgriffe\SyliusClerkPlugin\DataSyncInfrastructure\Enum\Resource;
+use Webgriffe\SyliusClerkPlugin\DataSyncInfrastructure\Model\QueryBuilderInterface;
+
+/**
+ * @implements QueryBuilderInterface<CustomerInterface>
+ */
+final readonly class CustomersQueryBuilder implements QueryBuilderInterface
+{
+    public function __construct(
+        private CustomerRepository $customerRepository,
+        private EventDispatcherInterface $eventDispatcher,
+    ) {
+    }
+
+    public function getResource(): Resource
+    {
+        return Resource::CUSTOMERS;
+    }
+
+    public function getResult(
+        ChannelInterface $channel,
+        string $localeCode,
+        ?\DateTimeInterface $modifiedAfter = null,
+        ?int $limit = null,
+        ?int $offset = null,
+    ): array {
+        $queryBuilder = $this->customerRepository->createQueryBuilder('c');
```

Comment:
Mh oki!

Reply:
But I'm not so sure 😂. Maybe it is better to retrieve them and make failing the normalizer? What do you think?

Reply:
Ok, good point, but if we let the normalizer fails it will probably log a warning (in the future) or, generally, it will throw an exception... but we already know here that this is not an exception, so we should avoid to let it pass.

For example, if the email of the customer would be nullable on Sylius (while it is mandatory on Clerk), we would definitely filter out customers without email here, no?

Reply:
Ok!